### PR TITLE
Adding container-interop compatibility

### PR DESCRIPTION
**Pull request by moufmouf**
Hey!

This pull request is a proposal to add "[container-interop](https://github.com/container-interop/container-interop)" compatibility to "config".
As you can see, this pull request is... wooh... 3 lines long :)

Config already complies with container-interop's `ContainerInterface`. The only thing we have to do is add the interface.

Why would we want to do that?

Container-interop is a common interface for dependency injection containers. Containers usually store services, but they can also store **parameters**. container-interop does not do any distinction between services and parameters. In container-interop vocabulary, containers are used to store "entries" (they can be either services or parameters... or whatever!)

Config could therefore be considered as a container providing config parameters. And this is absolutely great, because this means that any container compatible with container-interop (and there are a bunch of them!) would be able to fetch data from the Config container (one of the big ideas of container-interop is to allow a container to fetch parameters/service from other containers through [delegate dependency lookup](https://github.com/container-interop/container-interop/blob/master/docs/Delegate-lookup-meta.md))

In short, this pull request makes it possible to use Config really easily with a hug range of different projects.

Disclaimer: I'm one of container-interop authors, and I'm also the editor of [PSR-11](https://github.com/php-fig/fig-standards/blob/master/proposed/container.md) (with @mnapoli) that is currently in discussion and planning to transform container-interop into a PSR.

### CODE REVIEW

**Comment by hassankhan** [comment]
I don't think currently we implement the `has()` method as required by the interface - not a massive deal as we can reuse `offsetExists()`, I'm happy to merge this as long as everything works and test coverage is maintained :+1:

**Comment by moufmouf** [comment]
Actually, there is already a `has` method [here](https://github.com/hassankhan/config/blob/develop/src/AbstractConfig.php#L110-L113)

It is in the develop branch but not yet in the master branch.

I don't really see how I could improve the unit tests since I'm not adding any meaningful line of code, but if you have a suggestion, I'll be happy to write the test for it.

**Comment by moufmouf** [comment]
Ooops... wait just a bit more before merging this! I've just realized there is some more work to do regarding error handling. Actually, according to container-interop, the `get` method should throw an exception if no entry is found. But the current behaviour of `AbstractConfig` is to return `null`.

We could:

1- Change the behaviour of `get` to throw an exception (simplest approach but this is kind of a breaking change)
2- or I could instead write a wrapper around the `AbstractConfig` class that implements the `ContainerInterface`

What do you think?

**Comment by hassankhan** [comment]
Ah, I was looking in the `master` branch, my bad. If it already exists, then there should be tests for it. 

I think this might be something else to line up with a 1.x release, along with #46 - only because of the breaking change.

**Comment by ashnazg** [comment]
I'm trying to think of how a config _file_ itself would represent an object that I'd want a container to find for me.  Since looking at this library, I've assumed that only primitives are representable, with arrays being the most complicated item.  Granted, I realize that `set()` could store objects for the duration of execution, but I don't see a way for a container-held object to originate from a config file... and the library doesn't persist things back to file, so the container use case could not be to serialize the object to a string for storage back to a config file.

**Comment by hassankhan** [comment]
My understanding was this would be more useful in a subclass implementing `AbstractConfig` which would be set up manually.

**Comment by hassankhan** [comment]
Closing the PR ... this project's goal never really was to be a DI container :)
